### ISSUE TITLE
3D support

### ISSUE THREAD

**Issue by renxwise** [KlugerLab/FIt-SNE#66]
Current FIt-SNE only supports 1D and 2D reduction, will the future FIt-SNE support 3D reduction? 3D reduction is also appealing.

**Comment by linqiaozhi** [comment]
Good question; we discussed it a bit [here](https://github.com/KlugerLab/FIt-SNE/issues/34#issuecomment-419729622).

At this time, we are not planning on supporting 3-dimensional embedding. Our interpolation scheme would work for 3 dimensions, and still be O(N), but it the constant would be worse (i.e. many more interpolation nodes would be required). We have not tried it, so we do not have any benchmarks.

**Comment by dkobak** [comment]
3D embeddings are good for only one purpose: to create spinning and "fly-through" animations for public talks :-)

**Comment by renxwise** [comment]
@dkobak It's true for visualization. But the applications of tSNE should not be limited to visualization. It belongs to the wide field of dimensionality reduction.

**Comment by dkobak** [comment]
@renxwise Hmm. Can you give me any example of tSNE meaningfully/successfully used not for visualisation?

**Comment by renxwise** [comment]
@dkobak I am now working on a project to extend the scope of t-SNE application. When the manuscript is ready, I'll share with you.

**Comment by dkobak** [comment]
Yes, please do post a link here once it's ready. Sounds interesting.

**Comment by renxwise** [comment]
@dkobak @linqiaozhi  I post a manuscript in biorxiv which shows an application of t-SNE to infer 3D cell spatial organizations based on scRNA-seq data (https://t.co/AmLzVHkomp?amp=1).  I wrote the code with Matlab. But a faster tool is needed to handle larger numbers of cells. Welcome to give your comments.
